Add E2E tests.

whchung opened the pull request:
An assortment of tests previously failing are now passing. Add them into
regression test suite.

https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/40
https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/41
https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/114

Configs which are still failing are marked with FIXME.

whchung committed:
Add E2E tests.

An assortment of tests previously failing are now passing. Add them into
regression test suite.

yiqian1 commented on mlir/test/mlir-miopen-driver/e2e/conv2d_host_validation.mlir:
```diff
@@ -148,3 +148,50 @@
+mlir-miopen-driver -pv -fil_layout=yxck -in_layout=nhwc -out_layout=nhwk -batchsize=128 -in_channels=8 -out_channels=128 -in_h=16 -in_w=16 -fil_h=5 -fil_w=5 --padding_h=1 --padding_w=1 -p=false -c | mlir-rocm-runner --shared-libs=%rocm_wrapper_library_dir/librocm-runtime-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils.so --entry-point-result=void | FileCheck %s --check-prefix=CHECK_IS ...
+
+// Various dilation values
+mlir-miopen-driver -pv -fil_layout=yxck -in_layout=nhwc -out_layout=nhwk -batchsize=128 -in_channels=8 -out_channels=128 -in_h=16 -in_w=16 -fil_h=3 -fil_w=3 --dilation_h=2 --dilation_w=2 -p=false -c | mlir-rocm-runner --shared-libs=%rocm_wrapper_library_dir/librocm-runtime-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils.so --entry-point-result=void | FileCheck %s --check-prefix=CHECK_ ...
+mlir-miopen-driver -pv -fil_layout=yxck -in_layout=nhwc -out_layout=nhwk -batchsize=128 -in_channels=8 -out_channels=128 -in_h=16 -in_w=16 -fil_h=5 -fil_w=5 --dilation_h=2 --dilation_w=2 -p=false -c | mlir-rocm-runner --shared-libs=%rocm_wrapper_library_dir/librocm-runtime-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils.so --entry-point-result=void | FileCheck %s --check-prefix=CHECK_ ...
```
These two tests are already included in line 12 and 13.

whchung replied:
@yiqian1 I've revised the commit because they didn't have `RUN:` macro. Could you help identify these 2 duplicated configs once again? Thanks.

yiqian1 replied:
Yes. They are duplicated with line 12 and 13.

whchung replied:
@yiqian1 please help specify the duplicating line numbers for the new test cases so I know where to remove

yiqian1 replied:
Just remove line 169 and 170.

whchung committed:
Add E2E tests.

An assortment of tests previously failing are now passing. Add them into
regression test suite.

whchung committed:
Add E2E tests.

An assortment of tests previously failing are now passing. Add them into
regression test suite.

Configs which are still failing are marked as FIXME.

whchung committed:
Fix typo.

whchung committed:
Add missing f16 flags.

yiqian1 commented on mlir/test/mlir-miopen-driver/e2e/conv2d_host_validation_f16.mlir:
```diff
@@ -206,5 +206,5 @@
-// FIXME: mlir-miopen-driver -pv -fil_layout=yxck -in_layout=nhwc -out_layout=nhwk -batchsize=128 -in_channels=8 -out_channels=128 -in_h=32 -in_w=32 -fil_h=5 -fil_w=5 --padding_h=1 --padding_w=1 -p=false -c | mlir-rocm-runner --shared-libs=%rocm_wrapper_library_dir/librocm-runtime-wrappers%shextlib,%linalg_test_lib_dir/libmlir_runner_utils%shextlib --entry-point-result=void | FileCheck %s --check ...
-// FIXME: mlir-miopen-driver -pv -fil_layout=yxck -in_layout=nhwc -out_layout=nhwk -batchsize=64 -in_channels=8 -out_channels=128 -in_h=16 -in_w=64 -fil_h=3 -fil_w=5 -p=false -c | mlir-rocm-runner --shared-libs=%rocm_wrapper_library_dir/librocm-runtime-wrappers%shextlib,%linalg_test_lib_dir/libmlir_runner_utils%shextlib --entry-point-result=void | FileCheck %s --check-prefix=CHECK_ISSUE_40_5
+// FIXME: mlir-miopen-driver -pv -fil_layout=yxck -in_layout=nhwc -out_layout=nhwk -batchsize=128 -in_channels=8 -out_channels=128 -in_h=32 -in_w=32 -fil_h=5 -fil_w=5 --padding_h=1 --padding_w=1 -p=false -c | mlir-rocm-runner --shared-libs=%rocm_wrapper_library_dir/librocm-runtime-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils%shlibext --entry-point-result=void | FileCheck %s --check ...
+// FIXME: mlir-miopen-driver -pv -fil_layout=yxck -in_layout=nhwc -out_layout=nhwk -batchsize=64 -in_channels=8 -out_channels=128 -in_h=16 -in_w=64 -fil_h=3 -fil_w=5 -p=false -c | mlir-rocm-runner --shared-libs=%rocm_wrapper_library_dir/librocm-runtime-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils%shlibext --entry-point-result=void | FileCheck %s --check-prefix=CHECK_ISSUE_40_5
```
Add "-t f16".

whchung committed:
Remove duplicated test configs.

whchung commented:
There seems to be some flakiness in certain fp16 configs. Will run the test suite a couple more iterations to identify them and mark them as FIXME. There seems to be no flakiness on gfx908 node I use though.

whchung committed:
Mark one test config as FLAKY.
